watchdog.go

```diff
@@ -0,0 +1,91 @@
+package veneur
+
+import (
+	"sync/atomic"
+	"time"
+
+	rtdebug "runtime/debug"
+
+	"github.com/sirupsen/logrus"
+)
+
+type watchdogTickerMetadata struct {
+	Name           string
+	Ticker         *time.Ticker
+	StuckIntervals int
+	Interval       time.Duration
+}
+
+// FlushWatchdog periodically checks that at most
+// `flush_watchdog_missed_flushes` were skipped in a Server. If more
+// than that number was skipped, it panics (assuming that flushing is
+// stuck) with a full level of detail on that panic's backtraces.
+//
+// It never terminates, so is ideally run from a goroutine in a
+// program's main function.
+func (s *Server) FlushWatchdog() {
```

Comment:
I moved this into its own file because it seemed ideologically distinct from the other stuff that happens in server.go. We'll want a lot more automated testing here to make sure we're testing out the edge cases. If anything looks off here, lmk.